src/Features/Core/Portable/FullyQualify/AbstractFullyQualifyCodeFixProvider.cs

```diff
@@ -2,314 +2,61 @@
-                    var member = container.GetMembers(name).FirstOrDefault();
-                    memberName = member != null ? member.Name : name;
-                }
-                else
-                {
-                    memberName = name;
-                }
-
-                var title = $"{containerName}.{memberName}";
-                var codeAction = CodeAction.Create(
-                    title,
-                    cancellationToken => ProcessNodeAsync(document, simpleName, containerName, symbolResult.OriginalSymbol, cancellationToken),
-                    title);
-
-                yield return codeAction;
-            }
-        }
-
-        private async Task<Document> ProcessNodeAsync(Document document, TSimpleNameSyntax simpleName, string containerName, INamespaceOrTypeSymbol originalSymbol, CancellationToken cancellationToken)
-        {
-            var newRoot = await ReplaceNodeAsync(simpleName, containerName, originalSymbol.IsType, cancellationToken).ConfigureAwait(false);
-            return document.WithSyntaxRoot(newRoot);
-        }
-
-        private static bool IsValidNamedTypeSearchResult(
-            SemanticModel semanticModel, int arity, bool inAttributeContext,
-            bool looksGeneric, INamedTypeSymbol searchResult)
-        {
-            if (arity != 0 && searchResult.GetArity() != arity)
-            {
-                // If the user supplied type arguments, then the search result has to match the 
-                // number provided.
-                return false;
-            }
-
-            if (looksGeneric && searchResult.TypeArguments.Length == 0)
-            {
-                return false;
-            }
-
-            if (!searchResult.IsAccessibleWithin(semanticModel.Compilation.Assembly))
-            {
-                // Search result has to be accessible from our current location.
-                return false;
-            }
-
-            if (inAttributeContext && !searchResult.IsAttribute())
-            {
-                // If we need an attribute, we have to have found an attribute.
-                return false;
-            }
+            var service = document.GetRequiredLanguageService<IFullyQualifyService>();
+            var optFixData = await service.GetFixDataAsync(document, context.Span, hideAdvancedMembers, cancellationToken).ConfigureAwait(false);
+            if (optFixData is null)
+                return;
 
-            if (!HasValidContainer(searchResult))
-            {
-                // Named type we find must be in a namespace, or a non-generic type.
-                return false;
-            }
+            var fixData = optFixData.Value;
+            if (fixData.IndividualFixData.Length == 0)
+                return;
 
-            return true;
-        }
+            var codeActions = fixData.IndividualFixData.SelectAsArray(
+                d => CodeAction.Create(
+                    d.Title,
+                    async cancellationToken =>
```

Comment:
Before, the syntax tree replacement and edits were computed in resolve, now it looks like we compute the edits upfront.  I just want to confirm that it doesn't matter (it looks like a fairly inexpensive operation to compute the edits)

Reply:
Correct. This is what we do with add-using as well. The fix portion is cheap for this one. The expensive work is the symbolic work.